.github/workflows/main.yml

```diff
@@ -1,119 +1,127 @@
-        echo "URL=$url" >> "$GITHUB_ENV"
-
-    - name: "integration(wfb): save the pull request event trigger commit URL"
-      if: "contains(github.event_name, 'pull_request')"
-      run: |
-        url=${{ github.event.pull_request.url }}
-        echo "URL=$url" >> "$GITHUB_ENV"
-
-    - name: "integration(wfb): send a payload via workflow builder webhook"
-      id: slackWorkflow
-      uses: ./
-      with:
-        # Workflow builder webhooks need to know the name of the keys in the payload in advance. Without normalizing, the github context payload keys can differ based on the GitHub trigger event type
-        # Normalized payload with info pulled out from GitHub trigger event
-        payload: "{\"author\":\"${{ github.event.sender.login }}\",\"url\":\"${{ env.URL}}\", \"repoName\":\"${{ github.event.repository.full_name }}\", \"status\":\"${{ job.status }}\"}"
-      env:
-        SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
-
-    - name: "integration(wfb): confirm a payload was sent"
-      run: test -n "${{ steps.slackWorkflow.outputs.time }}"
-
-    - name: "integration(incoming): post a message via incoming webhook"
-      id: slackIncoming
-      uses: ./
-      with:
-        payload: "{\"text\":\"Incoming Webhook test for slack send\", \"blocks\":[{\"type\":\"section\",\"text\":{\"type\":\"plain_text\",\"text\":\"A post by Slack Send GitHub Action. Testing Incoming webhooks\",\"emoji\":true}}]}"
-      env:
-        SLACK_WEBHOOK_URL: ${{ secrets.SLACK_INCOMING_WEBHOOK_URL }}
-        SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
-
-    - name: "integration(incoming): confirm a webhook was posted"
-      run: test -n "${{ steps.slackIncoming.outputs.time }}"
-
-    - name: "integration(incoming): reveal contents of the github payload"
-      run: echo $JSON
-      env:
-        JSON: ${{ toJSON(github) }}
-
-    - name: "integration(incoming): post a message via payload file"
-      id: slackPayloadFile
-      uses: ./
-      with:
-        payload-file-path: ./.github/resources/payload-notification.json
-      env:
-        SLACK_WEBHOOK_URL: ${{ secrets.SLACK_INCOMING_WEBHOOK_URL }}
-        SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
-        JOB_STATUS: ${{ job.status }}
-        ATTACHMENT_COLOR: ${{ (job.status == 'success' && 'good') || (job.status == 'failure' && 'danger') || 'warning' }}
-
-    - name: "integration(incoming): confirm a payload file was posted"
-      run: test -n "${{ steps.slackPayloadFile.outputs.time }}"
+      - name: "build: checkout the latest changes"
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: "build: install the required dependencies"
+        run: npm ci
+
+      - name: "build: setup the linter for formatting checks"
+        uses: biomejs/setup-biome@v2
+        with:
+          version: latest
+
+      - name: "unit(test): perform lints and formatting checks"
+        run: biome ci .
```

Comment:
Also not so thrilled about this change, but found it helpful for fixing CI- It seemed like `biome` wasn't being downloaded in the step prior beforehand? 🤔 

<details>

https://github.com/slackapi/slack-github-action/actions/runs/10639670258/job/29498086533#step:4:13

```
  npm run lint
  shell: /usr/bin/bash -e {0}

> slack-github-action@2.0.0-development lint
> biome check

node:internal/modules/cjs/loader:1143
  throw err;
  ^

Error: Cannot find module '@biomejs/cli-linux-x64/biome'
Require stack:
- /home/runner/work/slack-github-action/slack-github-action/node_modules/@biomejs/biome/bin/biome
    at Module._resolveFilename (node:internal/modules/cjs/loader:1140:15)
    at Function.resolve (node:internal/modules/helpers:188:19)
    at Object.<anonymous> (/home/runner/work/slack-github-action/slack-github-action/node_modules/@biomejs/biome/bin/biome:51:11)
    at Module._compile (node:internal/modules/cjs/loader:1364:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Module._load (node:internal/modules/cjs/loader:1019:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:128:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/runner/work/slack-github-action/slack-github-action/node_modules/@biomejs/biome/bin/biome'
  ]
}

Node.js v18.20.4
```

</details>